test/test_namedtensor.py

```diff
@@ -1062,12 +1062,38 @@ def test_flatten(self):
             tensor.flatten(['H', 'D', 'W'], 'features')
 
     def test_unflatten(self):
-        tensor = torch.randn(7, 2 * 3 * 5, 11, names=('N', 'D', 'K'))
+        # test args: tensor, int, namedshape
+        self.assertTrue(torch.equal(
+            torch.ones(4).unflatten(0, (('A', 2), ('B', 2))),
+            torch.ones(2, 2, names=('A', 'B'))))
+        self.assertTrue(torch.equal(
+            torch.ones(4).unflatten(0, [('A', 2), ('B', 2)]),
+            torch.ones(2, 2, names=('A', 'B'))))
+        self.assertTrue(torch.equal(
+            torch.ones(4).unflatten(0, (['A', 2], ['B', 2])),
+            torch.ones(2, 2, names=('A', 'B'))))
+        self.assertTrue(torch.equal(
+            torch.ones(4).unflatten(-1, (['A', 2], ['B', 2])),
+            torch.ones(2, 2, names=('A', 'B'))))
+
+        # test args: namedtensor, int, namedshape
+        self.assertTrue(torch.equal(
+            torch.ones(2, 4, names=('A', 'B')).unflatten(1, (('B1', 2), ('B2', 2))),
+            torch.ones(2, 2, 2, names=('A', 'B1', 'B2'))))
+
+        # test args: namedtensor, str, namedshape
+        self.assertTrue(torch.equal(
+            torch.ones(2, 4, names=('A', 'B')).unflatten('B', (('B1', 2), ('B2', 2))),
+            torch.ones(2, 2, 2, names=('A', 'B1', 'B2'))))
+
+        # test invalid args: namedtensor, str, sizes
+        self.assertRaises(TypeError, lambda: torch.tensor([1], names=('A',)).unflatten('A', (1, 1)))
```

Comment:
nit: it's nice to use self.assertRaisesRegex to ensure that the error message is human readable